db/migrate/20150606065713_add_sti_to_lookup_keys.rb

```diff
@@ -0,0 +1,22 @@
+class AddStiToLookupKeys < ActiveRecord::Migration
+  def self.up
+    add_column :lookup_keys, :type, :string
+    LookupKey.scoped.each do |key|
+      if key.is_param
+        key.type = 'PuppetclassLookupKey'
+      else
+        key.type = 'VariableLookupKey'
+      end
+      key.save!
+    end
+    add_index :lookup_keys, :type
+
+    remove_foreign_key :environment_classes, :lookup_keys
```

Comment:
Add a conditional to check for presence of the foreign key before attempting removal, see other recent migrations (ptables, IIRC).